front/app/community/[siren]/components/FicheMarchesPublics/Treemap.tsx

```diff
@@ -0,0 +1,305 @@
+          y={leaf.y0 + 22}
+          fontSize={16}
+          fontWeight={700}
+          fill='white'
+          className='pointer-events-none'
+        >
+          {formatNumber(leaf.data.value)}
+        </text>
+      )}
+      {leaf.x1 - leaf.x0 > 80 && leaf.y1 - leaf.y0 > 60 && (
+        <text
+          x={leaf.x0 + 8}
+          y={leaf.y0 + 42}
+          fontSize={14}
+          fontWeight={500}
+          fill='white'
+          className='pointer-events-none'
+        >
+          {wrapText(leaf.data.name, leaf.x1 - leaf.x0 - 16).map((line, i) => (
+            <tspan key={i} x={leaf.x0 + 8} dy={i === 0 ? 0 : 14}>
+              {line}
+            </tspan>
+          ))}
+        </text>
+      )}
+    </g>
+  ));
+
+  return (
+    <>
+      <div className='flex items-center justify-between'>
+        <div className='flex items-baseline gap-2'>
+          <h3 className='py-2 text-xl'>Répartition </h3>
+          <div className='flex items-baseline gap-2'>
+            <div
+              onClick={() => {
+                setTableDisplayed(false);
+                setLinesDisplayed(0);
+              }}
+              className={`cursor-pointer ${!tableDisplayed ? 'text-neutral-800' : 'text-neutral-400'}`}
+            >
+              (graphique
+            </div>{' '}
+            <Switch
+              checked={tableDisplayed}
+              onCheckedChange={() => {
+                setTableDisplayed((prev) => !prev);
+                setLinesDisplayed(0);
+              }}
+            />
+            <div
+              onClick={() => {
+                setTableDisplayed(true);
+                setLinesDisplayed(0);
+              }}
+              className={`cursor-pointer ${tableDisplayed ? 'text-neutral-800' : 'text-neutral-400'}`}
+            >
+              tableau)
+            </div>{' '}
+          </div>
+        </div>
+        <div className='flex items-center gap-2'>
+          <YearSelector years={availableYears} onSelect={setSelectedYear} />
+          <DownloadSelector />
+        </div>
+      </div>
+      {tableDisplayed && (
+        <SectorTable
+          topSectors={topSectors}
+          formattedData={formattedData}
+          linesDisplayed={linesDisplayed}
+          onLoadMore={() => setLinesDisplayed((prev) => prev + 1)}
+        />
+      )}
+      {!tableDisplayed && (
+        <div ref={containerRef}>
+          <svg width={width} height={height}>
+            {allShapes}
+          </svg>
+          {tooltip.visible && (
+            <div
+              className='pointer-events-none fixed z-50 max-w-[200px] rounded bg-gray-900 px-3 py-2 text-sm text-white shadow-lg'
+              style={{
+                top: tooltip.y,
+                left: tooltip.x,
+                transform: 'translateY(-50%)',
+              }}
+            >
+              <div className='font-bold'>{tooltip.name}</div>
+              <div>{tooltip.value.toLocaleString()} €</div>
+            </div>
```

Comment:
Tu peux creer un composnat dans un fichier separe puis l utiliser

```
type TooltipProps = {
    visible: boolean;
    x: number;
    y: number;
    name: string;
    value: number;
  }
  
  export function Tooltip({x, y, name, value}: TooltipProps){
           return <div
              className='pointer-events-none fixed z-50 max-w-[200px] rounded bg-gray-900 px-3 py-2 text-sm text-white shadow-lg'
              style={{
                top: y,
                left: x,
                transform: 'translateY(-50%)',
              }}
            >
              <div className='font-bold'>{name}</div>
              <div>{value} €</div>
            </div>)}
```

```suggestions
<Tooltip {...tooltip}/>
```

Reply:
DOne